sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -166,7 +166,7 @@ object Utils {
     // It is safe to call head at here since functionsWithDistinct has at least one
     // AggregateExpression2.
     val distinctColumnExpressions =
-      functionsWithDistinct.head.aggregateFunction.children
+      functionsWithDistinct.flatMap(_.aggregateFunction.children)
```

Comment:
Why is this change necessary? Also, the comment above this line is now out-of-date.